modules/migration-configuring-mcg.adoc

```diff
@@ -0,0 +1,171 @@
+
+.Procedure
+
+. Log in to the CAM tool cluster.
+. Create the `NooBaa` CR configuration file, `noobaa.yml`, with the following content:
++
+[source,yaml]
+----
+apiVersion: noobaa.io/v1alpha1
+kind: NooBaa
+metadata:
+  name: noobaa
+  namespace: openshift-storage
+spec:
+ dbResources:
+   requests:
+     cpu: 0.5 <1>
+     memory: 1Gi
+ coreResources:
+   requests:
+     cpu: 0.5 <1>
+     memory: 1Gi
+----
+<1> For a very small cluster, you can change the `cpu` value to `0.1`.
+
+. Create the `NooBaa` object:
++
+----
+$ oc create -f noobaa.yml
+----
+
+. Create the `BackingStore` CR configuration file, `bs.yml`, with the following content:
++
+[source,yaml]
+----
+apiVersion: noobaa.io/v1alpha1
+kind: BackingStore
+metadata:
+  finalizers:
+  - noobaa.io/finalizer
+  labels:
+    app: noobaa
+  name: mcg-pv-pool-bs
+  namespace: openshift-storage
+spec:
+  pvPool:
+    numVolumes: 3 <1>
+    resources:
+      requests:
+        storage: 50Gi <2>
+    storageClass: gp2 <3>
+  type: pv-pool
+----
+<1> Specify the number of volumes in the PV pool.
+<2> Specify the size of the volumes.
+<3> Specify the storage class.
+
+. Create the `BackingStore` object:
++
+----
+$ oc create -f bs.yml
+----
+
+. Create the `BucketClass` CR configuration file, `bc.yml`, with the following content:
++
+[source,yaml]
+----
+apiVersion: noobaa.io/v1alpha1
+kind: BucketClass
+metadata:
+  labels:
+    app: noobaa
+  name: mcg-pv-pool-bc
+  namespace: openshift-storage
+spec:
+  placementPolicy:
+    tiers:
+    - backingStores:
+      - mcg-pv-pool-bs
+      placement: Spread
+----
+
+. Create the `BucketClass` object:
++
+----
+$ oc create -f bc.yml
+----
+
+. Create the `ObjectBucketClaim` CR configuration file, `obc.yml`, with the following content:
++
+[source,yaml]
+----
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: migstorage
+  namespace: openshift-storage
+spec:
+  bucketName: migstorage <1>
+  storageClassName: openshift-storage.noobaa.io
+  additionalConfig:
+    bucketclass: mcg-pv-pool-bc
+----
+<1> Record the bucket name for adding the replication repository to the CAM web console.
+
+. Create the `ObjectBucketClaim` object:
++
+----
+$ oc create -f obc.yml
+----
+
+. Watch the resource creation process to verify that the `ObjectBucketClaim` status is `Bound`:
++
+----
+$ watch -n 30 'oc get -n openshift-storage objectbucketclaim migstorage -o yaml'
+----
++
+This process can take five to ten minutes.
+
+. Obtain and record the following values, which are required when you add the replication repository to the CAM web console:
+
+* S3 endpoint:
++
+----
+$ oc get route -n openshift-storage s3
+----
+
+* S3 provider access key:
++
+----
+$ oc get secret -n openshift-storage migstorage -o go-template='{{ .data.AWS_ACCESS_KEY_ID }}' | base64 -d
+----
+
+* S3 provider secret access key:
++
+----
+$ oc get secret -n openshift-storage migstorage -o go-template='{{ .data.AWS_SECRET_ACCESS_KEY }}' | base64 -d
+----
```

Comment:
Above steps just clarify how to creating the MCG storage bucket, but it doesn't mention how to configure the MCG storage bucket as a replication repository. Especially, what bucket name should be set? You could refer to https://github.com/jmontleon/blogpost/blob/master/cam-ocs-pvpool/README.md

Reply:
That procedure is in a different section, https://migration-configuring-mcg-storage--ocpdocs.netlify.com/openshift-enterprise/latest/migration/migrating_4_1_4/migrating-applications-with-cam.html#migration-adding-replication-repository-to-cam_migrating-4_1-4_x

The bucket name can be any valid name (we don't need to tell the user all the constraints on names; we assume the user knows that the operation will fail if an invalid parameter is entered). 

The user specifies the bucket name in line 132 of this file and records it, to use when adding MCG to the CAM.